src/hardware/input/mouseif_dos_driver.cpp

```diff
@@ -920,8 +889,120 @@ void MOUSEDOS_AfterNewVideoMode(const bool setmode)
 	state.update_region_y[1] = -1; // offscreen
 	state.cursor_type        = MouseCursor::Software;
 	state.enabled            = true;
+	state.inhibit_draw       = false;
 
-	clear_pending_events();
+	// Some software (like 'Down by the Laituri' game) is known to first set
+	// the min/max mouse cursor position and then set VESA mode, therefore
+	// (unless this is a driver reset) skip setting min/max position and
+	// granularity for SVGA graphic modes
+
+	if (is_mode_set && is_svga_mode && !is_svga_text) {
+		return;
+	}
```

Comment:
Regarding min/max positions, I have a suspicion that the Dostodon client that crashes when moving the mouse out-of-bounds is also having some min/max issues. Are there reasonable or sane min/max values that can be set on a video mode change, so even if the software doesn't set these up right, they still get some protection based on the video mode's bounds?

Reply:
If I remember correctly, Dostodon was crashing even when using CtMouse driver. At this concrete point we shouldn't change anything; SVGA text/graphic modes are handled at the end of the function.